### PR TITLE
Tests now use pyftpdlib to run.

### PR DESCRIPTION
**Pull request by morris25**
Also a few small changes have been made to the interface to make the function signatures match up with each other better.

### CODE REVIEW

File: src/FTPC.jl [hunk 1]
```diff
@@ -266,6 +263,13 @@ function ftp_get(ctxt::ConnContext, file_name::AbstractString, save_path::Abstra
         p_wd = pointer_from_objref(wd)
         p_resp = pointer_from_objref(resp)
 
+        @ce_curl curl_easy_setopt CURLOPT_VERBOSE Int64(1)
+
+        # Force active mode
+        @ce_curl curl_easy_setopt CURLOPT_FTP_USE_EPSV 0
+        @ce_curl curl_easy_setopt CURLOPT_FTP_USE_EPRT 0
+        @ce_curl curl_easy_setopt CURLOPT_FTPPORT "-"
```
Comment (omus) [inline src/FTPC.jl]: Verbose option was just meant for debugging and should be removed. Comment out the "force active mode" code as it was just meant for testing. We probably want to support that functionality later.

File: src/FTPC.jl [hunk 2]
```diff
@@ -303,25 +307,21 @@ function ftp_get(ctxt::ConnContext, file_name::AbstractString, save_path::Abstra
         if ~isempty(save_path)
             close(wd.buffer)
         end
+
+        @ce_curl curl_easy_setopt CURLOPT_VERBOSE Int64(0)
```
Comment (omus) [inline src/FTPC.jl]: Also remove

File: src/FTPC.jl [hunk 3]
```diff
@@ -210,32 +210,32 @@ end
...
 
-@doc """
-Download file with non-persistent connection.
+"""
+    ftp_get(options::RequestOptions, file_name::AbstractString, save_path::AbstractString=""; mode::FTP_MODE=binary_mode)
 
-- url: FTP server, ex "localhost"
-- file_name: name of file to download
-- options: options for connection, ex use ssl, implicit security, etc.
-- save_path: location to save file to, if not specified file is written to a buffer
```
Comment (omus) [inline src/FTPC.jl:234]: Some of these arguments are still relevant and could be documented. See #4 in http://julia.readthedocs.io/en/latest/manual/documentation/#documentation

File: src/FTPObject.jl [hunk 4]
```diff
@@ -1,12 +1,19 @@
 type FTP
     ctxt::ConnContext
 
-    function FTP(;host::AbstractString="", implicit::Bool=false, ssl::Bool=false,
-            verify::Bool=true, active::Bool=false, user::AbstractString="",
-            pswd::AbstractString="")
+    """
+        FTP(;hostname::AbstractString="", implicit::Bool=false, ssl::Bool=false,
+            verify_peer::Bool=true, active_mode::Bool=false, username::AbstractString="",
+            password::AbstractString="")
```
Comment (omus) [inline src/FTPObject.jl]: Probably best to describe the arguments below and just use `FTP(; kwargs...) -> FTP` as the signature

File: src/FTPObject.jl [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
+
+    Create FTPObject.
```
Comment (omus) [inline src/FTPObject.jl]: "Create an `FTP` object"

File: src/FTPC.jl (hunk 1 above)
Comment (morris25) [inline src/FTPC.jl]: I was wondering why ftp_get was always in active mode. That explains things.

File: test/ftp_object.jl [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+ftp = FTP(; opts...)
+no_unexpected_changes(ftp)
+Base.close(ftp)
+
+# check readdir
+ftp = FTP(; opts...)
+server_dir = readdir(ftp)
+contains(string(server_dir), "test_directory")
+contains(string(server_dir), "test_byte_file")
+contains(string(server_dir), "test_download.txt")
```
Comment (omus) [inline test/ftp_object.jl]: Missing `@test`?

File: test/ssl.jl [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    test_persistent_upload(ctxt)
+
+    # the connection has to be closed during test_persistent_upload to get the server file to write out
+    # ftp_close_connection(ctxt)
+
+    close(server)
+end
+
+function test_download(options)
+    download_file = "test_download.txt"
```
Comment (omus) [inline test/ssl.jl]: Rename "download_file" to "server_path"

File: test/ssl.jl [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    resp = ftp_get(options, download_file)
+    @test resp.code == 226
+    @test readstring(resp.body) == readstring(joinpath(ROOT, download_file))
+end
+
+function test_upload(options)
+    local_file = "test_upload.txt"
+    server_file = joinpath(ROOT, local_file)
```
Comment (omus) [inline test/ssl.jl]: Rename "local_file" to "client_path" and "server_file" to "local_server_path"

File: test/ssl.jl [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+function test_persistent_download(ctxt)
+    resp = ftp_command(ctxt, "PWD")
+    @test resp.code == 257
+    @test readstring(resp.body) == ""
+end
+
+function test_persistent_upload(ctxt)
+    upload_file = "test_upload.txt"
+    server_file = joinpath(ROOT, upload_file)
```
Comment (omus) [inline test/ssl.jl]: Rename "upload_file" to "client_path" and "server_file" to "local_server_path"

File: test/ssl.jl [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    resp = ftp_command(options, "PWD")
+    @test resp.code == 257
+    @test readstring(resp.body) == ""
+end
+
+function test_persistent_download(ctxt)
+    resp = ftp_command(ctxt, "PWD")
+    @test resp.code == 257
+    @test readstring(resp.body) == ""
+end
```
Comment (omus) [inline test/ssl.jl:81]: Doesn't actually test downloading

File: test/ssl.jl [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    @test resp.code ==226
+    @test readstring(upload_file) == readstring(server_file)
+    rm(server_file)
+end
+
+function test_persistent_cmd(ctxt)
+    resp = ftp_command(ctxt, "PWD")
+    @test resp.code == 257
+    @test readstring(resp.body) == ""
+end
```
Comment (omus) [inline test/ssl.jl:99]: You could use multiple dispatch here more effectively:

```
test_cmd(options:: RequestOptions) = ...
test_cmd(ctxt::ConnContext) = ...
```

File: test/utils.jl [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        expected = string("^\\Q", replace(expected, "...", "\\E.*\\Q"), "\\E\$")
+        expected = replace(expected, "\\Q\\E", "")  # Remove empty quoted sections
+        return ismatch(Regex(expected), original)
+    else
+        return original == expected
+    end
+end
+
+function cleanup_file(filename::AbstractString)
+    if (isfile(filename))
```
Comment (omus) [inline test/utils.jl]: Drop the extra parentheses. `if isfile(filename)`

File: test/utils.jl [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        rm(filename)
+    end
+    @test !isfile(filename)
+end
+
+function cleanup_dir(dirname::AbstractString, recursive = true)
+    if (isdir(dirname))
```
Comment (omus) [inline test/utils.jl]: Remove extra parentheses

File: test/non_ssl.jl [hunk 14]
```diff
@@ -0,0 +1,302 @@
...
+
+function test_command(headers, opt)
+    resp = ftp_command(opt, "PWD")
+    check_response(resp, "", 257, headers)
+end
+
+function tests_by_mode(active::Bool)
+
+    options = RequestOptions(; opts..., ssl=false, active_mode=active)
+    mode_header = active? "200 Active data connection established." : "229 Entering extended passive mode (...)."
```
Comment (omus) [inline test/non_ssl.jl]: Add a space after "active"

File: test/non_ssl.jl [hunk 15]
```diff
@@ -0,0 +1,302 @@
...
+)
+
+# options = RequestOptions(hostname="127.0.0.1:8000", username="user", passwd="passwd", ssl=false, active_mode=true)
+function check_response(resp, body, code, headers)
+    @test typeof(resp.total_time) == Float64
+    @test resp.code == code
+    @test readstring(resp.body) == body
+    @test resp.bytes_recd == length(body)
+    @test is_headers_equal(resp.headers, headers)
+end
```
Comment (omus) [inline test/non_ssl.jl:28]: Rename to `test_response`?

File: test/non_ssl.jl [hunk 16]
```diff
@@ -0,0 +1,302 @@
...
+    resp = ftp_get(options, local_file, save_path)
+    server_file = joinpath(ROOT, local_file)
+    body = readstring(server_file)
+
+    @test isfile(save_path) == true
+
+    @test readstring(save_path) == body
+    @test resp.code == 226
+    @test readstring(resp.body) == ""
+    @test is_headers_equal(resp.headers, headers)
```
Comment (omus) [inline test/non_ssl.jl]: Isn't this just `check_response`?

File: test/ssl.jl (hunk 10 above)
Comment (morris25) [inline test/ssl.jl:81]: I think I accidentally copied over something there, I'll have to check that out.

It was supposed to be testing ftp_get with ctxt. Thanks for catching that!

File: test/server/python_deps.jl [hunk 17]
```diff
@@ -0,0 +1,32 @@
+# this script uses the default Conda settings to get python, pyopenssl, and pyftpdlib
+# if you already have an acceptable version of python and pip installed it will attempt to use those instead
+
+using Conda
+using Compat
+
+py_version(python::AbstractString) = chomp(readstring(`$python -c "import platform; print(platform.python_version())"`))
```
Comment (omus) [inline test/server/python_deps.jl]: Why do this over `$python --version`?

File: test/server/python_deps.jl [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+const python = try
+    py = get(ENV, "PYTHON", isfile("PYTHON") ? readchomp("PYTHON") : "python")
+    version = convert(VersionNumber, py_version(py))
```
Comment (omus) [inline test/server/python_deps.jl]: Should be `version = VersionNumber(py_version(py))`

File: test/server/python_deps.jl [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+    py
+catch error
+    info( "No system-wide Python was found; got the following error:\n",
+          "$error\nusing the Python distribution in the Conda package")
+    Conda.add("pyopenssl")
+    abspath(Conda.PYTHONDIR, "python" * ( is_windows() ? ".exe" : ""))
+end
+
+pip = (dirname(python) == abspath(Conda.PYTHONDIR))? abspath(Conda.SCRIPTDIR, "pip" * ( is_windows() ? ".exe" : "")): "pip"
+run(`$pip install pyftpdlib`)
```
Comment (omus) [inline test/server/python_deps.jl]: What's going on here? You're using Conda pip to install pyftpdlib but not pyopenssl?

Comment (omus) [inline test/server/python_deps.jl]: Are there any problems with using Conda for everything?

File: test/server/server.py [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+        cert.gmtime_adj_notBefore(0)
+        cert.gmtime_adj_notAfter(10*365*24*60*60)
+        cert.set_issuer(cert.get_subject())
+        cert.set_pubkey(k)
+        cert.sign(k, 'sha1')
+
+        open(join(cert_dir, cert_file), "wt").write(
+            crypto.dump_certificate(crypto.FILETYPE_PEM, cert))
+        open(join(cert_dir, key_file), "wt").write(
+            crypto.dump_privatekey(crypto.FILETYPE_PEM, k))
```
Comment (omus) [inline test/server/server.py]: Cleaner and safer to write as:

```
with open(join(cert_dir, key_file), "wt") as fp:
    fp.write(crypto.dump_privatekey(crypto.FILETYPE_PEM, k))
```

File: test/server/server.py [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+    parser.add_argument('--hostname', type=str, default="localhost")
+    parser.add_argument('--port', type=int, default=0)
+    parser.add_argument('--passive-ports', type=str)
+    parser.add_argument('--tls', choices=['implicit', 'explicit'])
+    parser.add_argument('--tls-require', choices=['control', 'data'], nargs='*', default=[])  # noqa
+    parser.add_argument('--cert-file', type=str, default='test.crt')
+    parser.add_argument('--key-file', type=str, default='test.key')
+    parser.add_argument('--cert-dir', type=str, default='')
+    parser.add_argument('--debug', type=bool, default=False)
+    parser.add_argument('--gen-certs', type=bool, default=False)
```
Comment (omus) [inline test/server/server.py]: Since `--cert-dir` is only used by `--gen-certs` why not combine and make `--gen-certs-dir`

File: appveyor.yml [hunk 22]
```diff
@@ -27,7 +27,7 @@ install:
 build_script:
 # Need to convert from shallow to complete for Pkg.clone to work
   - IF EXIST .git\shallow (git fetch --unshallow)
-  - C:\projects\julia\bin\julia -e "versioninfo(); Pkg.init(); Pkg.add(\"LibCURL\"); Pkg.checkout(\"LibCURL\"); Pkg.clone(\"https://github.com/samuel-massinon-invenia/JavaCall.jl.git\"); Pkg.checkout(\"JavaCall\", \"pull-request/bf8b4987\"); Pkg.clone(pwd(), \"FTPClient\"); Pkg.build(\"FTPClient\");"
+  - C:\projects\julia\bin\julia -e "versioninfo(); Pkg.init(); Pkg.add(\"LibCURL\"); Pkg.checkout(\"LibCURL\"); Pkg.clone(pwd(), \"FTPClient\"); Pkg.build(\"FTPClient\");"
```
Comment (omus) [inline appveyor.yml]: Don't forget to remove the clone from the .travis.yml file as well.

Comment (omus) [inline appveyor.yml]: Totally not your fault but adding LibCURL should be handled by the REQUIRE file. Remove the section:

```
Pkg.init(); Pkg.add(\"LibCURL\"); Pkg.checkout(\"LibCURL\"); 
```

You'll also want to make this change to the .travis.yml file. Make this in a separate commit than the change above.

File: .gitignore [hunk 23]
```diff
@@ -6,5 +6,9 @@ deps/ext
 test/MockFTPServerJulia.class
```
Comment (omus) [inline .gitignore]: Both `deps/ext` and `test/MockFTPServerJulia.class` can be removed from this file.

File: README.md [hunk 24]
```diff
@@ -93,17 +90,15 @@ rm(ftp::FTP, file_name::AbstractString)
...
 ### Examples
 
-Using non-peristent connection and FTPS with implicit security:
+Using non-persistent connection and FTPS with implicit security:
 ```julia
 using FTPClient
 
 ftp_init()
-options = RequestOptions(ssl=true, implicit=true, username="user1", passwd="1234", url="localhost")
+options = RequestOptions(ssl=true, implicit=true, username="user1", password="1234", url="localhost")
```
Comment (omus) [inline README.md]: Switch "url" to "hostname" here

File: README.md [hunk 25]
```diff
@@ -150,7 +145,7 @@ ftp_cleanup()
 Using the FTP object with a persistent connection and FTPS with implicit security:
 ```julia
 ftp_init()
-ftp = FTP(host="localhost", implt=true, ssl=true, user="user3", pswd="2468" )
+ftp = FTP(host="localhost", implicit=true, ssl=true, username="user3", password="2468" )
```
Comment (omus) [inline README.md]: You forgot to change `host` to `hostname` here.

File: src/FTPC.jl [hunk 26]
```diff
@@ -155,9 +155,9 @@ function setup_easy_handle(options::RequestOptions)
     @ce_curl curl_easy_setopt CURLOPT_URL options.url
     # @ce_curl curl_easy_setopt CURLOPT_VERBOSE Int64(1)
 
-    if (~isempty(options.username) && ~isempty(options.passwd))
+    if (~isempty(options.username) && ~isempty(options.password))
```
Comment (omus) [inline src/FTPC.jl]: Remove extra parentheses and switch `~` to `!` as `~` is bitwise negation.

File: test/non_ssl.jl (hunk 16 above)
Comment (morris25) [inline test/non_ssl.jl]: My previous response was incorrect, the comparison did not work because resp.body is empty when it writes to a file. The number of bytes received must equal the length of the file instead of the instead of the length of the resp.body.
Sorry I didn't recall that correctly the first time.

File: test/server/python_deps.jl (hunk 17 above)
Comment (morris25) [inline test/server/python_deps.jl]: It isolates the version number instead of having to use a regex to grab it after the python call. I'll probably end up changing it later.
The result from this is "2.7.11" instead of "Python 2.7.11" on my computer.

File: test/server/python_deps.jl (hunk 19 above)
Comment (morris25) [inline test/server/python_deps.jl]: I'll have to double check to be sure, but I believe the issue I was having was that pyftpdlib wasn't available through Conda, but pyopenssl was. 
Using Conda to install pyopenssl triggers it to install the latest python 2.7 and pip into the Conda deps directory as part of the installation of miniconda. This is what Pycall does as well. It is by no means the best solution, but it does install the right version of python into a predictable place on all platforms. It gets everything up and running until I can find a better solution.

Comment (morris25) [inline test/server/python_deps.jl]: I can now confirm that there is no osx version of pyftpdlib available through Conda at present. There are packages available through separate channels for Linux and Windows, and I can look into what is required to get a channel up for an osx version. Until then this solution works for all platforms at once.

File: src/FTPC.jl [hunk 27]
```diff
@@ -351,6 +402,8 @@ function ftp_put(ctxt::ConnContext, file_name::AbstractString, file::IO; mode::F
     p_rd = pointer_from_objref(rd)
     p_resp = pointer_from_objref(resp)
 
+    @ce_curl curl_easy_setopt CURLOPT_VERBOSE Int64(1)
+
```
Comment (omus) [inline src/FTPC.jl]: More left over verbosity options. Should be removed.

File: test/ftp_object.jl [hunk 28]
```diff
@@ -0,0 +1,251 @@
...
+    Base.close(ftp)
+end
+
+# check connection error
+@test_throws FTPClientError FTP(hostname="not a host", username=username(server), password=password(server), ssl=false)
+
+# check object
+ftp = FTP(; opts...)
+no_unexpected_changes(ftp)
+Base.close(ftp)
```
Comment (omus) [inline test/ftp_object.jl]: Why always `Base.close`? You should be able to just do `close`

File: test/server/server.jl [hunk 29]
```diff
@@ -0,0 +1,92 @@
...
+
+Conda.add_channel("invenia")
+Conda.add("pyftpdlib")
+
+const ROOT = abspath(dirname(@__FILE__), "root")
+const SCRIPT = abspath(dirname(@__FILE__), "server.py")
+const CERT = abspath(dirname(@__FILE__), "test.crt")
+const KEY = abspath(dirname(@__FILE__), "test.key")
+
+python =
```
Comment (omus) [inline test/server/server.jl]: Mistake?

**Comment by omus** [comment]
@morris25 were all the original tests transferred? If not can you point out which tests were removed and why?

File: test/server/server.jl (hunk 29 above)
Comment (morris25) [inline test/server/server.jl]: yup, I may have gotten distracted....

File: test/non_ssl.jl [hunk 30]
```diff
@@ -0,0 +1,308 @@
...
+    ctxt, resp = ftp_connect(options)
+    test_put(headers, ctxt)
+
+end
+
+tests_by_mode(true)
+tests_by_mode(false)
+
+# test binary vs ascii
+@unix_only upload_bytes = string("466F6F426172", "0A", "466F6F426172")
```
Comment (omus) [inline test/non_ssl.jl]: `@unix_only` is deprecated (I forgot). Switch to `is_unix() && (upload_bytes = ...)`

**Comment by morris25** [comment]
There were some tests that checked error situations. Since we're not using a mock server anymore there isn't a built-in way to get the server to block connections, so some of the previous test_client_errors tests got jettisoned until I find a way around that.

File: test/ftp_object.jl (hunk 28 above)
Comment (morris25) [inline test/ftp_object.jl]: close is being odd

File: test/non_ssl.jl [hunk 31]
```diff
@@ -0,0 +1,308 @@
...
+open(joinpath(ROOT, byte_file), "w") do fp
+    write(fp, hex2bytes(download_bytes))
+end
+
+# it is not the same file when downloading in ascii mode
+options = RequestOptions(; opts..., ssl=false, active_mode=false)
+resp = ftp_get(options, byte_file; mode=ascii_mode)
+bytes = read(resp.body)
+is_unix() && (@test bytes != hex2bytes(download_bytes))
+is_unix() && (@test bytes == hex2bytes(download_bytes_ascii))
```
Comment (omus) [inline test/non_ssl.jl]: Parentheses around last statement can be dropped here. Only was needed for assignment to work.

File: test/non_ssl.jl [hunk 32]
```diff
@@ -0,0 +1,308 @@
...
+resp = ftp_get(options, byte_file)
+bytes = read(resp.body)
+@test bytes == hex2bytes(download_bytes)
+
+# it is not the same file when downloading in ascii mode
+ctxt, resp = ftp_connect(options)
+resp = ftp_get(ctxt, byte_file, mode=ascii_mode)
+bytes = read(resp.body)
+is_unix() && (@test bytes != hex2bytes(download_bytes))
+is_unix() && (@test bytes == hex2bytes(download_bytes_ascii))
```
Comment (omus) [inline test/non_ssl.jl]: Drop extra parentheses

File: test/non_ssl.jl [hunk 33]
```diff
@@ -0,0 +1,308 @@
...
+bytes = read(resp.body)
+@test bytes == hex2bytes(download_bytes)
+ftp_close_connection(ctxt)
+
+# it is not the same file when downloading in ascii mode
+ftp = FTP(; opts...)
+buff = download(ftp, byte_file, mode=ascii_mode)
+bytes = read(buff)
+is_unix() && (@test bytes != hex2bytes(download_bytes))
+is_unix() && (@test bytes == hex2bytes(download_bytes_ascii))
```
Comment (omus) [inline test/non_ssl.jl]: Drop extra parentheses

File: test/non_ssl.jl [hunk 34]
```diff
@@ -0,0 +1,308 @@
...
+bytes = read(buff)
+is_unix() && @test bytes != hex2bytes(download_bytes)
+is_unix() && @test bytes == hex2bytes(download_bytes_ascii)
+buff = download(ftp, byte_file)
+bytes = read(buff)
+@test bytes == hex2bytes(download_bytes)
+buff = download(ftp, byte_file, mode=ascii_mode)
+bytes = read(buff)
+is_unix() && (@test bytes != hex2bytes(download_bytes))
+is_unix() && (@test bytes == hex2bytes(download_bytes_ascii))
```
Comment (omus) [inline test/non_ssl.jl]: Drop extra parentheses

File: test/runtests.jl [hunk 35]
```diff
@@ -1,200 +1,32 @@
...
+cleanup_file(upload_file)
+cleanup_file(joinpath(ROOT, download_file))
 
-end
+ftp_cleanup()
+teardown_server()
 
-# Done testing
-rm(upload_file_name)
+close(server)
```
Comment (omus) [inline test/runtests.jl]: If a test fails then the server won't be shutdown. You should wrap most of this in a try/finally:

``` julia
server = FTPServer()
try
    ...
finally
    close(server)
end
```

File: test/ssl.jl [hunk 36]
```diff
@@ -0,0 +1,100 @@
...
+    @test resp.code == 226
+
+    test_cmd(ctxt)
+    test_download(ctxt)
+    test_upload(ctxt)
+
+    # the connection has to be closed during test_upload(ctxt) to get the server file to write out
+    # ftp_close_connection(ctxt)
+
+    close(server)
```
Comment (omus) [inline test/ssl.jl]: Also needs to be wrapped in a try/finally

File: .gitignore [hunk 37]
```diff
@@ -2,9 +2,12 @@
 *.jl.mem
 
 # Ignore files we can build.
-deps/ext
-test/MockFTPServerJulia.class
+test/PYTHON
```
Comment (omus) [inline .gitignore]: Should be removed?

Comment (morris25) [inline .gitignore]: yup

**Comment by morris25** [comment]
oops, pushed the wrong thing

File: test/runtests.jl [hunk 38]
```diff
@@ -1,200 +1,35 @@
...
-    end
+    include("ssl.jl")
+    include("non_ssl.jl")
+    include("ftp_object.jl")
 
-    stop_server()
-    JavaCall.destroy()
+finally
+    cleanup_file(upload_file)
+    cleanup_file(joinpath(ROOT, download_file))
```
Comment (omus) [inline test/runtests.jl:28]: Both `upload_file` and `download_file` need to be defined before the try

Comment (morris25) [inline test/runtests.jl:28]: Yeah, I forgot to commit that before pushing

File: test/utils.jl [hunk 39]
```diff
@@ -0,0 +1,49 @@
...
+function cleanup_dir(dirname::AbstractString, recursive = true)
+    if isdir(dirname)
+        rm(dirname)
+    end
+    @test !isdir(dirname)
+end
+
+macro test_headers(original, expected)
+
+end
```
Comment (omus) [inline test/utils.jl]: Should be removed.

File: test/ssl.jl [hunk 40]
```diff
@@ -0,0 +1,109 @@
...
+        # ftp_close_connection(ctxt)
+    finally
+        close(server)
+
+        for line in eachline(server.io)
+            print(line)
+            if contains(line, "FTP session closed")
+                break
+            end
+        end
```
Comment (omus) [inline test/ssl.jl]: Would be good to leave a comment indicating why this is here.

Comment (morris25) [inline test/ssl.jl]: Actually the output from the server isn't really that useful, so I think I'm going to just remove it.

**Comment by omus** [comment]
Going to merge now. There are still some issues with LibCURL failing on Windows but we'll sort that out in another PR.
